content/docs/cucumber/state.md

```diff
@@ -112,10 +125,12 @@ The available dependency injection modules are:
 - [Guice](#guice)
 - [OpenEJB](#openejb)
 - [Weld](#weld)
-- [Needle](#needle)
+- [Needle](#needle) 
 
-### PicoContainer
+{{% /block %}} {{% block "ruby,javascript" %}} Dependency Injection is specific to JVM languages. {{% /block %}}
```

Comment:
So for JS and Ruby, there is no "dependency injection", so that list of dependency injection modules just has the names and the verbiage about it being specific to JVM languages. It took me a long time to figure this out! I'm sure coders understand when they read this. It just seemed odd to list them out when they aren't even available in those languages.

Reply:
There reason for this is technical; we made it so we cannot have headers inside a language block (although I forgot the reason we did this). Which means that we cannot "hide" the headers that are irrelevant for a particular language, unfortunately. Hence the generic text saying something is not relevant.